src/http-requests.ts

```diff
@@ -166,6 +171,39 @@ class HttpRequests {
     }
   }
 
+  async fetchWithTimeout(
+    url: string,
+    options: RequestInit | undefined,
```

Comment:
RequestInit is proper to `fetch` I think. What happens if the user decides use `axios` for example as `this.httpClient`?

Reply:
Yes, it will break but adding the `Record<string, any>` to the `options` parameter seems to be the best solution since we don't have access to `axios` types.